src/defs.h

```diff
@@ -108,6 +112,26 @@ typedef enum {
 
 /* variable definition */
 typedef struct {
+    int counter;
+    int stack[64];
+    int stack_idx;
+} rename_t;
+
+struct BB;
```

Comment:
To maintain a consistent coding style, I would recommend using `basic_block_t` instead of the overly simplified `BB`.

See https://github.com/icxd/llvm-ir/blob/main/lib/llvm.h

Reply:
By the way, it is certainly acceptable to name an instance of `basic_block_t` as `bb` for brevity.